File: LeetCode/Easy/0506-relative-ranks/0506-relative-ranks.java
class Solution {
    public String[] findRelativeRanks(int[] score) {
        int[] sorted = score.clone();
        Arrays.sort(sorted);

        String[] ans = new String[score.length];
        int[] ranked = new int[sorted[sorted.length - 1] + 1];

        for(int i = 0; i < sorted.length; ++i) {
            ranked[sorted[i]] = sorted.length - 1 - i;
        }

        for(int i = 0; i < score.length; ++i) {
            if(ranked[score[i]] >= 3) ans[i] = Integer.toString(ranked[score[i]] + 1);
            else {
                if (ranked[score[i]] == 0) ans[i] = "Gold Medal";
                else if (ranked[score[i]] == 1) ans[i] = "Silver Medal";
                else if (ranked[score[i]] == 2) ans[i] = "Bronze Medal";
            }
        }

        return ans;
    }
}

File: LeetCode/Easy/0506-relative-ranks/0506-relative-ranks.kt
class Solution {

    fun findRelativeRanks(score: IntArray): Array<String> {

        val sorted: IntArray = score.clone();
        sorted.sortDescending();

        val ans: Array<String> = Array(score.size){""}
        val ranked: IntArray = IntArray(sorted[0] + 1)

        for((idx, no) in sorted.withIndex()) {
            ranked[no] = idx
        }

        for((idx, no) in score.withIndex()) {
            if(ranked[no] >= 3) ans[idx] = (ranked[no] + 1).toString()
            else {
                if(ranked[no] == 0) ans[idx] = "Gold Medal"
                else if(ranked[no] == 1) ans[idx] = "Silver Medal"
                else if(ranked[no] == 2) ans[idx] = "Bronze Medal"
            }
        }

        return ans
        
    }

}